tree/ntuple/v7/src/RField.cxx

```diff
@@ -1794,6 +1803,37 @@ void ROOT::Experimental::RClassField::ReadInClusterImpl(RClusterIndex clusterInd
    }
 }
 
+void ROOT::Experimental::RClassField::BeforeConnectPageSource(Internal::RPageSource &pageSource)
+{
+   if (GetOnDiskId() == kInvalidDescriptorId) {
+      return;
+   }
+   // Gather all known sub fields in the descriptor.
+   std::unordered_set<std::string> knownSubFields;
```

Comment:
@enirolf I *think* this may be the place where we can load sub fields that are used as targets for IO read rules. Together with allowing rules on missing fields, this should be a first step towards manual schema evolution...